src/lib.rs

```diff
@@ -30,6 +30,9 @@ pub trait OpenatDirExt {
     /// of course.
     fn open_file_optional<P: openat::AsPath>(&self, p: P) -> io::Result<Option<fs::File>>;
 
+    /// Remove a file from the given directory.
+    fn remove_file_optional<P: openat::AsPath>(&self, p: P) -> io::Result<()>;
+
```

Comment:
I think the docstring forgot to cover the `_optional` part of this.

@cgwalters should this method (and possibly a few other similar ones below) return a `Result<bool>` or something like that to let the caller distinguish the "action performed" and "nothing to be done" cases?

Reply:
Hmm, yeah no harm in returning the bool.  Would changing existing methods count as a semver break?  I suppose not.  Should be a separate PR anyways.

Reply:
It should be an API break because it changes trait signature, yes (but please double-check my words with https://github.com/rust-dev-tools/rust-semverver). No hurry to do it now, I've recorded it at https://github.com/cgwalters/openat-ext/issues/7 so that it can be done later on (outside of the hotpath of bootupd).